### ISSUE TITLE
option to disable utf8-validation

### ISSUE THREAD

**Issue by vinniefalco** [boostorg/beast#1201]
This would be even faster than the currently optimized version of `utf8_checker`.

**Comment by carolinebeltran** [comment]
Hi Vinnie, is this a feature that you are thinking of adding to Beast?  If so, I think that it would be a good idea, for instance, I use ICU for UTF-8 validation, comparison, transformation and any manipulation required such as word boundaries, etc.  I see no point in redundant validation for users who may use other Unicode libraries.

**Comment by vinniefalco** [comment]
Yes this is a planned feature. Note, you can always use binary mode for messages, no checking is performed then.

**Comment by alandefreitas** [comment]
Would that be another 

```cpp
void
set_option(utf8_option const& opt)
```

?

Do we want the logic in `stream`, checking if we should ignore the checker, or in `utf8_checker`, with some internal logic to disable it?

**Comment by vinniefalco** [comment]
```
void set_validate_utf8( bool v );
bool validate_utf8() const noexcept;
```

logic in `stream`
